llama_index/llms/bedrock_utils.py

```diff
@@ -61,38 +61,98 @@
     "meta.llama2-13b-chat-v1",
 }
 
-# Each bedrock model specifies parameters with a slightly different name
-# most of these are passed optionally by the user in kwargs but max tokens
-# is a required argument.
-PROVIDER_SPECIFIC_PARAM_NAME = {
-    "amazon": {"max_tokens": "maxTokenCount"},
-    "ai21": {"max_tokens": "maxTokens"},
-    "anthropic": {"max_tokens": "max_tokens_to_sample"},
-    "cohere": {"max_tokens": "max_tokens"},
-    "meta": {"max_tokens": "max_gen_len"},
-}
 
-# The response format for each provider is different
-PROVIDER_RESPONSE_LOADER = {
-    "amazon": lambda x: x["results"][0]["outputText"],
-    "ai21": lambda x: x["completions"][0]["data"]["text"],
-    "anthropic": lambda x: x["completion"],
-    "cohere": lambda x: x["generations"][0]["text"],
-    "meta": lambda x: x["generation"],
-}
+class Provider(ABC):
```

Comment:
Clever! But also, the fact that bedrock does not have a consistent interface is INSANE 🙃

Reply:
Yes. They just relay whatever API the downstream provider exposes without modification.